x-pack/legacy/plugins/canvas/server/plugin.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup, PluginsSetup } from './shim';
+import { routes } from './routes';
+import { functions } from '../canvas_plugin_src/functions/server';
+import { registerCanvasUsageCollector } from './usage';
+import { loadSampleData } from './sample_data';
+
+export class Plugin {
+  public setup(core: CoreSetup, plugins: PluginsSetup) {
+    routes(core);
+
+    const { serverFunctions } = plugins.interpreter.register({ serverFunctions: functions });
+
+    core.injectUiAppVars('canvas', async () => {
+      const config = core.getServerConfig();
+      const basePath = config.get('server.basePath');
+      const reportingBrowserType = (() => {
+        const configKey = 'xpack.reporting.capture.browser.type';
+        if (!config.has(configKey)) {
+          return null;
+        }
+        return config.get(configKey);
+      })();
+      const kibanaVars = await core.getInjectedUiAppVars('kibana');
```

Comment:
FYI this pattern will not exist in the New Platform. Instead, the Kibana plugin should include any of this values in it's setup contract. You could change this now to just pull these values directly from a kibana plugin and in your shim you could call `getInjectedUiAppVars` to populate the fake plugin interface.